docs/zdgbook/ObjectPublishing.rst

```diff
@@ -926,42 +926,46 @@ button will also deactivate all other radio buttons from the other records.
 Exceptions
 ----------
 
-Unhandled exceptions are caught by the object publisher and are
-translated automatically to nicely formatted HTTP output.
-
-When an exception is raised, the exception type is mapped to an HTTP
-code by matching the value of the exception type with a list of
-standard HTTP status names. Any exception types that do not match
-standard HTTP status names are mapped to "Internal Error" (500). The
-standard HTTP status names are: "OK", "Created", "Accepted", "No
-Content", "Multiple Choices", "Redirect", "Moved Permanently", "Moved
-Temporarily", "Not Modified", "Bad Request", "Unauthorized",
-"Forbidden", "Not Found", "Internal Error", "Not Implemented", "Bad
-Gateway", and "Service Unavailable". Variations on these names with
-different cases and without spaces are also valid.
-
-An attempt is made to use the exception value as the body of the
-returned response. The object publisher will examine the exception
-value. If the value is a string that contains some white space, then
-it will be used as the body of the return error message. If it
-appears to be HTML, the error content type will be set to 'text/html',
-otherwise, it will be set to 'text/plain'. If the exception value is
-not a string containing white space, then the object publisher will
-generate its own error message.
-
-There are two exceptions to the above rule:
-
-1. If the exception type is: "Redirect", "Multiple Choices" "Moved
-   Permanently", "Moved Temporarily", or "Not Modified", and the
-   exception value is an absolute URI, then no body will be provided
-   and a 'Location' header will be included in the output with the
-   given URI.
-
-2. If the exception type is "No Content", then no body will be
-   returned.
-
-When a body is returned, traceback information will be included in a
-comment in the output.
+When the object publisher catches an unhandled exception, it tries to
+match it with a set of predifined exceptions coming from the
+Products.zExceptions package, such as **HTTPNoContent**, **HTTPNotFound**,
+**HTTPUnauthorized** and many, many more. If there is a match, the custom
+exception gets upgraded into an zException, which then results in a proper
```

Comment:
upgraded **to a** zException

Instead of punting by just saying "many, many more" I would list them like the old documentation did.

Reply:
I removed the "many, many more" phrase and added a link to the package for a full list of exceptions..

The old documentation included more, but also only a partial list of all the possible exceptions.

Plus, if there will be a change to the list of exceptions of the zExceptions package (which probably won't ever happen), nobody will think about updating the Zope developer guide.

Also, if I really want to implement a custom exception I am probably more interested to choose the right status code. Choosing the name is only a way to get there. Either way, I'd have to lookup the source code of zExceptions anyway.